x-pack/plugins/infra/public/containers/logs/log_flyout.tsx

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import createContainer from 'constate-latest';
+import { isString } from 'lodash';
+import React, { useContext, useEffect, useMemo, useState } from 'react';
+import { FlyoutItemQuery, InfraLogItem } from '../../graphql/types';
+import { useApolloClient } from '../../utils/apollo_context';
+import { UrlStateContainer } from '../../utils/url_state';
+import { useTrackedPromise } from '../../utils/use_tracked_promise';
+import { flyoutItemQuery } from './flyout_item.gql_query';
+
+export enum FlyoutVisibility {
+  hidden = 'hidden',
+  visible = 'visible',
+}
+
+interface FlyoutOptionsUrlState {
+  flyoutId?: string | null;
+  flyoutVisibility?: string | null;
+  surroundingLogsId?: string | null;
+}
+
+export const useLogFlyout = ({ sourceId }: { sourceId: string }) => {
+  const [flyoutVisible, setFlyoutVisibility] = useState<boolean>(false);
+  const [flyoutId, setFlyoutId] = useState<string | null>(null);
+  const [flyoutItem, setFlyoutItem] = useState<InfraLogItem | null>(null);
+  const [surroundingLogsId, setSurroundingLogsId] = useState<string | null>(null);
+
+  const apolloClient = useApolloClient();
+
+  const [loadFlyoutItemRequest, loadFlyoutItem] = useTrackedPromise(
+    {
+      cancelPreviousOn: 'creation',
+      createPromise: async () => {
+        if (!apolloClient) {
+          throw new Error('Failed to load flyout item: No apollo client available.');
+        }
+
+        return await apolloClient.query<FlyoutItemQuery.Query, FlyoutItemQuery.Variables>({
+          fetchPolicy: 'no-cache',
+          query: flyoutItemQuery,
+          variables: {
+            itemId: (flyoutId != null && flyoutId) || '',
```

Comment:
I know this was just inherited from the previous implementation. :innocent: But given that we have more control over the request, how about we don't request anything at all if the `flyoutId` was not set? That case could just be handled early similar to the lack of an `apolloClient`. An empty string as the id just doesn't seem to make sense to me.